cmd/common.go

```diff
@@ -71,13 +74,59 @@ Fail:
 }
 
 func assertHoverInitialized() {
-	_, err := os.Stat("desktop")
+	_, err := os.Stat(buildPath)
 	if os.IsNotExist(err) {
-		fmt.Println("hover: Directory 'desktop' is missing, did you run `hover init` in this project?")
+		if hoverMigration() {
+			return
+		}
+		fmt.Println("hover: Directory '" + buildPath + "' is missing, did you run `hover init` in this project?")
 		os.Exit(1)
 	}
 	if err != nil {
 		fmt.Printf("hover: Failed to detect directory desktop: %v\n", err)
 		os.Exit(1)
 	}
 }
+
+func hoverMigration() bool {
+	oldBuildPath := "desktop"
+	file, err := os.Open(filepath.Join(oldBuildPath, "go.mod"))
+	if err != nil {
+		return false
+	}
+	defer file.Close()
+
+	fmt.Println("hover: ⚠ Found older hover directory layout, hover is now expecting a 'go' directory instead of 'desktop'.")
+	fmt.Println("hover: ⚠    To migrate, rename the 'desktop' directory to 'go'.")
+	fmt.Printf("hover:      Let hover do the migration? ")
+
+	if askForConfirmation() {
+		err := os.Rename(oldBuildPath, buildPath)
+		if err != nil {
+			fmt.Printf("hover: Migration failed: %v\n", err)
+			return false
+		}
+		fmt.Printf("hover: Migration success\n")
+		return true
+	}
+
+	return false
+}
+
+// askForConfirmation asks the user for confirmation.
+func askForConfirmation() bool {
+	fmt.Printf("[Y/n]: ")
```

Comment:
We may not want to default to "yes", this is an unexpected prompt (not happy flow) and the user may just ramble on their keyboard, hitting enter to see some log lines...

I once wrote a tiny package for these kinds of (t)asks, https://github.com/GeertJohan/go.ask
It's very old, and I would have named things differently now, but it works.. (also has built-in retries, etc.)
Replacing `if askForConfirmation(){` with `if ask.MustDefaultAsk(false, "hover:      Let hover do the migration?") {` should do the trick.

Reply:
Wait, nevermind, lets not use that. It depends on linenoise which is C which complicates `go get`-ing hover.